### PR TITLE
Added wishlist functionality without Auth

### PR DESCRIPTION
**Pull request by bhavyaj12**
### Wishlist Page with functionality

- I have added the wishlist component using useContext and useReducer
- I have modified the product component to enable wishlist functionality
- Full Wishlist page with functional remove button 
- Added mockman route

### Files to check

- [ ] src/components/navigation-top/NavigationTop.jsx
- [ ] src/components/product-card/ProductCard.jsx
- [ ] src/components/wishlist-card/WishlistCard.jsx
- [ ] src/contexts/wishlist-context.js
- [ ] src/pages/wishlist-page/WishlistPage.jsx
- [ ] src/reducers/wishlist-reducer.js
- [ ] src/utilities/filterFunctions.js

Ignore other files, they're index, styles, database and images

### What not to check

- Functionality of cart pages
- Move to cart button on wishlist card
- Authentication

image

### CODE REVIEW

**Review by vikrantkode — APPROVED** [review]
- Functionality is working as expected in UI
- Code is easily readable.
-

File: src/routes/Routes.jsx [hunk 1]
```diff
@@ -11,6 +12,12 @@ export default function SiteRoutes() {
             <Route path="/products" element={<ProductsPage />} />
             <Route path="/signup" element={<SignupPage />} />
             <Route path="/wishlist" element={<WishlistPage />} />
+            <Route path="/mock" element={
+                <Mockman>
+
+                </Mockman>
```
Comment (ravikumar0403) [inline src/routes/Routes.jsx:18]: This can be a self closing tag

File: src/components/product-card/ProductCard.jsx [hunk 2]
```diff
@@ -101,17 +107,41 @@ export default function ProductCard ({
...
               Add to Cart
             </span>
-          </a>
-          <a href="#" className="button btn-outline button-secondary ">
-            <span>
-              <i className="far fa-heart"></i>
-            </span>
-          </a>
+          </button>
+          {wishState.wishlist.find((prod) => prod._id === _id) ? (
```
Comment (ravikumar0403) [inline src/components/product-card/ProductCard.jsx:116]: You can extract out this logic as a separate common function  to check for both cart and wishlist by passing parameters accordingly

**Review by ravikumar0403 — APPROVED** [review]
- Awesome UI and the code is also clean.
- Really liked the brief description of the PR and the conventions followed in commit messages
- Folders are also well structured. Amazing work.

File: src/components/product-card/ProductCard.jsx (hunk 2 above)
Comment (bhavyaj12) [reply src/components/product-card/ProductCard.jsx:116]: Thanks Ravi, will note and Refactor at end!

**Pull request merged**